framework/test/unit/specs/modules/token/token_module.spec.ts

```diff
@@ -165,7 +266,100 @@ describe('token module', () => {
 			);
 		});
 	});
-	// TODO: Add #reducers test
-	// TODO: Add #afterBlockApply test
-	// TODO: Add #afterGenesisBlockApply test
+
+	describe('#afterBlockApply', () => {
+		const generatorPublicKey = getRandomBytes(20);
+		let block: any;
+		const minFee =
+			BigInt(genesisConfig.minFeePerByte) * BigInt(1) + BigInt(genesisConfig.baseFees[0].baseFee);
+		beforeEach(() => {
+			block = {
+				header: {
+					generatorPublicKey,
+					reward: BigInt(1),
+				},
+				payload: [
+					{
+						moduleType: 2,
+						assetType: 0,
+						getBytes: () => [1],
+						fee: BigInt(20000000),
+					},
+				],
+			};
+		});
+
+		describe('when block contains transactions', () => {
+			it('should update generator balance to give rewards and fees - minFee', async () => {
+				let expected = BigInt(senderAccount.token.balance) + block.header.reward;
```

Comment:
this should be generator balance?

Reply:
sender is the generator in the setup so it's correct

Reply:
generatorPublicKey is randomBytes, so im not sure how its getting this sender

Reply:
in the test setup, stateStore.account.get is mocked to return sender.

Reply:
ah I see. The state store is already a mock, so it shouldn't need to be stubbed. It can be used as it is.

Reply:
in the other module tests, it was written this way but I will update to just change the return value and not mock it.